plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractHiveConnectorTest.java

```diff
@@ -201,6 +204,11 @@ protected final QueryRunner createQueryRunner()
                 .setInitialTables(REQUIRED_TPCH_TABLES)
                 .build();
 
+        if (!exchangeManagerProperties.isEmpty()) {
```

Comment:
It feels to me that more elagant way would be to do it in the subclass which really cares about. Maybe we can
*  revert making `createQueryRunner` final, 
* drop `exchangeManagerProperties`
* In a subclass do:
```java
    @Override
    protected QueryRunner createQueryRunner()
            throws Exception
    {
        QueryRunner queryRunner = super.createQueryRunner();

        queryRunner.installPlugin(new FileSystemExchangePlugin());
        queryRunner.loadExchangeManager("filesystem", ...);
        
        return queryRunner;

    }
```

Reply:
I agree with you, one obstacle here is that when you have `setInitialTables`, the queryRunner will populate data, which already checks for exchange manager, so I can't`installPlugin` and `loadExchangeManager`. How do you think I should get around this?

Reply:
In fact, I agree with you. However, one obstacle here is that when you have `setInitialTables`, the queryRunner will populate data, which already checks for exchange manager, so I can't `installPlugin` and `loadExchangeManager` after the queryRunner has been constructed. How should I get around here?

Reply:
What about exposing 
```
    protected void fineTuneQueryRunner(HiveQueryRunner.Builder builder) {
    }
```
in `AbstractHiveConnectorTest`
Then we could drop `Map<String, String> extraProperties, Map<String, String> exchangeManagerProperties` from constructor parameters and do fine tuning in subcalsses.

Reply:
I will keep it as it is right now as `TestHiveFaultTolerantExecutionConnectorTest` is the only subclass that needs it at the moment.